src/msg/async/ProtocolV2.cc

```diff
@@ -1291,40 +1291,32 @@ CtPtr ProtocolV2::handle_message() {
 #endif
   recv_stamp = ceph_clock_now();
 
-  auto header_frame = MessageFrame::Decode(
-    std::move(rx_segments_data[SegmentIndex::Msg::HEADER]));
-  // XXX: paranoid copy just to avoid oops
-  ceph_msg_header2 current_header = header_frame.header();
-
-  ldout(cct, 20) << __func__
-		 << " got envelope type=" << current_header.type
-		 << " src " << peer_name
-		 << " off " << current_header.data_off
-                 << dendl;
-
-  INTERCEPT(16);
+  // we need to get the size before std::moving segments data
+  const size_t cur_msg_size = get_current_msg_size();
```

Comment:
`cur_msg_size` was leaking because of being called after `decode_message`. Confirmed via `ceph_assert(current_msg_size == current_msg_size_before`).